sql/telemetry_derived/asn_aggregates_v1/query.sql

```diff
@@ -0,0 +1,129 @@
+
+-- SELECT autonomous_system_number, network
+-- FROM `static.geoip2_isp_blocks_ipv4`
+-- WHERE NET.IP_TRUNC(NET.SAFE_IP_FROM_STRING("100.1.0.255"), CAST(SPLIT(network, "/")[OFFSET(1)] AS INT64)) = NET.SAFE_IP_FROM_STRING(SPLIT(network, "/")[OFFSET(0)])
+CREATE TEMPORARY FUNCTION get_client_ip(xff STRING, remote_address STRING, pipeline_proxy STRING)
+RETURNS STRING
+LANGUAGE js
+AS
+  """
+  // X-Forwarded-For is a list of IP addresses
+  if (xff != null) {
+    // Google's load balancer will append the immediate sending client IP and a global
+    // forwarding rule IP to any existing content in X-Forwarded-For as documented in:
+    // https://cloud.google.com/load-balancing/docs/https/#components
+    //
+    // In practice, many of the "first" addresses are bogus or internal,
+    // so we target the immediate sending client IP.
+    ips = xff.split(",");
+
+    if (pipeline_proxy != null) {
+      // Drop extra IP from X-Forwarded-For
+      ips = ips.slice(0, -1);
+    }
+
+    ip = ips[Math.max(ips.length - 2, 0)].trim();
+  } else {
+    ip = "";
+    if (remote_address != null) {
+      ip = remote_address;
+    }
+  }
+
+  return ip;
+  """;
+
+--
+WITH asn_ip_address_range AS (
+  SELECT
+    NET.SAFE_IP_FROM_STRING(SPLIT(network, "/")[OFFSET(0)]) AS network_ip,
+    CAST(SPLIT(network, "/")[OFFSET(1)] AS INT64) AS mask,
+    autonomous_system_number
+  FROM
+    `moz-fx-data-shared-prod.static.isp_blocks_ipv4_20200407`
```

Comment:
We got a temporary table with the ISP data in shared-prod. In the future, this table will be updated weekly and probably renamed to `isp_blocks_ipv4`.